### PR TITLE
Some improvements to rollup and ts config

### PR DESCRIPTION
**Pull request by darinmorrison**

### CODE REVIEW

**Comment by Veetaha** [comment]
Oh, we had some bad experience with that typescript plugin. I am not sure we should use it...
https://github.com/rust-analyzer/rust-analyzer/pull/3003

File: editors/code/rollup.config.js [hunk 1]
```diff
@@ -1,24 +1,27 @@
-import resolve from '@rollup/plugin-node-resolve';
 import commonjs from '@rollup/plugin-commonjs';
+import resolve from '@rollup/plugin-node-resolve';
+import typescript from '@rollup/plugin-typescript';
 import nodeBuiltins from 'builtin-modules';
 
-export default {
-    input: 'out/main.js',
+/** @type { import('rollup').RollupOptions } */
```
Comment (Veetaha) [inline editors/code/rollup.config.js]: Oh, this one is very neat!

Comment (Veetaha) [reply editors/code/rollup.config.js]: Though, I guess the only thing that is left this one guy as the first line in this file?
```ts
// @ts-check
```

File: editors/code/package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     },
     "enableProposedApi": true,
     "scripts": {
-        "vscode:prepublish": "tsc && rollup -c",
```
Comment (matklad) [inline editors/code/package.json]: Yeah, I'd rather stick with invoking tsc manually

**Comment by darinmorrison** [comment]
Okay, I can remove the use of `@rollup/plugin-typescript`. Will push a change soon.

File: editors/code/rollup.config.js (hunk 1 above)
Comment (darinmorrison) [reply editors/code/rollup.config.js]: In vscode you will get the typing on hover (and completion) without the explicit `// @ts-check`, but it's probably a good idea to add it anyway.

File: editors/code/tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "compilerOptions": {
         "module": "commonjs",
-        "target": "es2018",
+        "target": "es2019",
```
Comment (matklad) [inline editors/code/tsconfig.json]: Where do I look which es version is supported on node which is shipped with VS Code? Like, how can I check what's the highest target we can use here?

Comment (lnicola) [reply editors/code/tsconfig.json]: It doesn't seem clear-cut, but https://node.green/#ES2019.

**Comment by matklad** [comment]
bors r+

Thanks!

File: editors/code/tsconfig.json (hunk 3 above)
Comment (darinmorrison) [reply editors/code/tsconfig.json]: @matklad for reference, this is the table I was looking at: https://kangax.github.io/compat-table/es2016plus/#chrome78. You will have to "show obsolete platforms" checkbox in the top left.

**Pull request merged**